src/Essentials/src/MediaPicker/MediaPicker.ios.cs

```diff
@@ -132,62 +203,121 @@ static FileResult DictionaryToMediaFile(NSDictionary info)
 				assetUrl = info[UIImagePickerController.ImageUrl] as NSUrl;
 
 				// Try the MediaURL sometimes used for videos
-				if (assetUrl == null)
-					assetUrl = info[UIImagePickerController.MediaURL] as NSUrl;
+				assetUrl ??= info[UIImagePickerController.MediaURL] as NSUrl;
 
-				if (assetUrl != null)
+				if (assetUrl is not null)
 				{
 					if (!assetUrl.Scheme.Equals("assets-library", StringComparison.OrdinalIgnoreCase))
+					{
 						return new UIDocumentFileResult(assetUrl);
-#pragma warning disable CA1416 // TODO: 'UIImagePickerController.PHAsset' is only supported on: 'ios' from version 11.0 to 14.0
-#pragma warning disable CA1422 // Validate platform compatibility
+					}
+
 					phAsset = info.ValueForKey(UIImagePickerController.PHAsset) as PHAsset;
-#pragma warning restore CA1422 // Validate platform compatibility
-#pragma warning restore CA1416
 				}
 			}
 
 #if !MACCATALYST
-#pragma warning disable CA1416 // TODO: 'UIImagePickerController.ReferenceUrl' is unsupported on 'ios' 11.0 and later
-#pragma warning disable CA1422 // Validate platform compatibility
-			if (phAsset == null)
+			if (phAsset is null)
 			{
 				assetUrl = info[UIImagePickerController.ReferenceUrl] as NSUrl;
 
-				if (assetUrl != null)
-					phAsset = PHAsset.FetchAssets(new NSUrl[] { assetUrl }, null)?.LastObject as PHAsset;
+				if (assetUrl is not null)
+				{
+					phAsset = PHAsset.FetchAssets([assetUrl], null)?.LastObject as PHAsset;
+				}
 			}
-#pragma warning restore CA1422 // Validate platform compatibility
-#pragma warning restore CA1416 // 'PHAsset.FetchAssets(NSUrl[], PHFetchOptions?)' is unsupported on 'ios' 11.0 and later
 #endif
 
-			if (phAsset == null || assetUrl == null)
+			if (phAsset is null || assetUrl is null)
 			{
 				var img = info.ValueForKey(UIImagePickerController.OriginalImage) as UIImage;
 
-				if (img != null)
+				if (img is not null)
+				{
 					return new UIImageFileResult(img);
+				}
 			}
 
-			if (phAsset == null || assetUrl == null)
+			if (phAsset is null || assetUrl is null)
+			{
 				return null;
-#pragma warning disable CA1416 // https://github.com/xamarin/xamarin-macios/issues/14619
-#pragma warning disable CA1422 // Validate platform compatibility
+			}
+
 			string originalFilename = PHAssetResource.GetAssetResources(phAsset).FirstOrDefault()?.OriginalFilename;
-#pragma warning restore CA1422 // Validate platform compatibility
-#pragma warning restore CA1416
 			return new PHAssetFileResult(assetUrl, phAsset, originalFilename);
 		}
 
 		class PhotoPickerDelegate : UIImagePickerControllerDelegate
 		{
 			public Action<NSDictionary> CompletedHandler { get; set; }
 
-			public override void FinishedPickingMedia(UIImagePickerController picker, NSDictionary info) =>
+			public override void FinishedPickingMedia(UIImagePickerController picker, NSDictionary info)
+			{
+				picker.DismissViewController(true, null);
```

Comment:
Do we always want to dismiss it ? was it done by other method before?

Reply:
Yep we did that earlier as well: https://github.com/dotnet/maui/pull/28920/files#diff-eeccaf91d6640c88e049bc177cc42dd92f6d57a57b80589b3bfc7a1aa819eb60L89